### PR TITLE
The default button in FiltersConfigurationDialog is in the wrong section #85

### PR DESCRIPTION
**Pull request by eobrienPilz**
This fixes issue 85 https://github.com/eclipse-platform/eclipse.platform.ui/issues/85

### CODE REVIEW

**Review by mickaelistria — CHANGES_REQUESTED** [review]
The fix is not at the right place: the FilterConfigurationDialog should keep extending the ViewSettingsDialog as it makes a lot of sense; the issue of the "Restore Defaults" button being misplaced is something that is common to several other dialog; so either it's fixes in ViewSettingsDialog and this Dialog inherits the change, or this dialog is kept as it.

**Comment by mickaelistria** [comment]
I spent more time looking at this proposal and come to the conclusion that ViewSettingsDialog is of no help and has no strong semantic and is not much used. So moving away from this class is eventually a good thing; and such patch would be welcome.
Have you considered using a TrayDialog as base class instead of a plain Dialog?

**Comment by eobrienPilz** [comment]
> I spent more time looking at this proposal and come to the conclusion that ViewSettingsDialog is of no help and has no strong semantic and is not much used. So moving away from this class is eventually a good thing; and such patch would be welcome. Have you considered using a TrayDialog as base class instead of a plain Dialog?

Looking at the documentation for TrayDialog it looks to be correct to use it here. It has a help link appearing by default but this can be disabled for now. The option of the help link may be useful when fixing issue #86. I will push these changes now.

**Comment by vogella** [comment]
@mickaelistria you are blocking this PR. Any update you still require?

File: bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/views/markers/FiltersConfigurationDialog.java [hunk 1]
```diff
@@ -678,6 +691,13 @@ protected void okPressed() {
...
 
 	@Override
+	protected void buttonPressed(int buttonId) {
+		if (buttonId == DEFAULTS_BUTTON_ID) {
+			performDefaults();
+		}
+		super.buttonPressed(buttonId);
+	}
+
 	protected void performDefaults() {
```
Comment (mickaelistria) [inline bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/views/markers/FiltersConfigurationDialog.java:701]: `@Override` should remain here.

Comment (eobrienPilz) [reply bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/views/markers/FiltersConfigurationDialog.java:701]: Its not @Override now that its using TrayDialog.

Comment (mickaelistria) [reply bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/views/markers/FiltersConfigurationDialog.java:701]: Ok, my bad.

**Review by mickaelistria — APPROVED** [review]

**Comment by mickaelistria** [comment]
Thanks!

**Pull request merged**